File: academic-system/src/main/java/org/example/repository/RepositorioTurmaTxt.java
package org.example.repository;

import org.example.model.Turma;
import org.example.model.Avaliacao;
import java.util.List;
import java.io.FileWriter;
import java.io.PrintWriter;
import java.io.IOException;
import org.example.exception.ExcecaoSistemaAcademico;

public class RepositorioTurmaTxt implements RepositorioTurma {
    
    private String arquivo = "turmas.txt";

    public RepositorioTurmaTxt() {
    }

    public RepositorioTurmaTxt(String arquivo) {
        this.arquivo = arquivo;
    }

    @Override
    public void salvarTodas(List<Turma> turmas) {
        try (PrintWriter writer = new PrintWriter(new FileWriter(arquivo))) {
            for (Turma turma : turmas) {
                // AC4: The TXT file must contain at least: class code, class title, assessment type, assessment value, assessment weight.
                List<Avaliacao> avaliacoes = turma.getAvaliacoes();
                if (avaliacoes.isEmpty()) {
                    writer.printf("%s|%s|SemAvaliacao|0.0|0.0%n",
                        turma.getCodigo(),
                        turma.getTitulo()
                    );
                } else {
                    for (Avaliacao avaliacao : avaliacoes) {
                        writer.printf("%s|%s|%s|%.2f|%.2f%n",
                            turma.getCodigo(),
                            turma.getTitulo(),
                            avaliacao.getClass().getSimpleName(),
                            avaliacao.getValor(),
                            avaliacao.getPeso()
                        );
                    }
                }
            }
        } catch (IOException e) {
            throw new ExcecaoSistemaAcademico("Erro ao salvar dados no arquivo TXT: " + e.getMessage());
        }
    }
}

File: academic-system-csharp/SistemaAcademico/src/SistemaAcademico/Repositories/RepositorioTurmaTxt.cs
using SistemaAcademico.Models;
using SistemaAcademico.Exceptions;

namespace SistemaAcademico.Repositories;

// Equivalente à classe Java: RepositorioTurmaTxt implements RepositorioTurma
// java.io.FileWriter / PrintWriter → System.IO.StreamWriter
// writer.printf → string.Format / ToString com InvariantCulture
// Padrão de projeto: Repository + Strategy (preservados)
public class RepositorioTurmaTxt : IRepositorioTurma
{
    private string arquivo = "turmas.txt";

    public RepositorioTurmaTxt() { }

    public RepositorioTurmaTxt(string arquivo)
    {
        this.arquivo = arquivo;
    }

    public void SalvarTodas(List<Turma> turmas)
    {
        try
        {
            using StreamWriter writer = new StreamWriter(arquivo);
            foreach (Turma turma in turmas)
            {
                // AC4: O arquivo TXT deve conter: código, título, tipo, valor, peso
                List<Avaliacao> avaliacoes = turma.GetAvaliacoes();
                if (avaliacoes.Count == 0)
                {
                    writer.WriteLine($"{turma.Codigo}|{turma.Titulo}|SemAvaliacao|0.0|0.0");
                }
                else
                {
                    foreach (Avaliacao avaliacao in avaliacoes)
                    {
                        // Equivalente ao writer.printf("%s|%s|%s|%.2f|%.2f%n", ...)
                        writer.WriteLine(string.Format(
                            System.Globalization.CultureInfo.InvariantCulture,
                            "{0}|{1}|{2}|{3:F2}|{4:F2}",
                            turma.Codigo,
                            turma.Titulo,
                            avaliacao.GetType().Name,
                            avaliacao.Valor,
                            avaliacao.Peso
                        ));
                    }
                }
            }
        }
        catch (IOException e)
        {
            throw new ExcecaoSistemaAcademico("Erro ao salvar dados no arquivo TXT: " + e.Message);
        }
    }
}
